### ISSUE TITLE
Providing different results of FLOPS calculation when using tf.compat.v1.profiler.profile on Mac M1 chip and on NVIDIA 1650 

### ISSUE THREAD

**Issue by shujaat81** [tensorflow/tensorflow#53785]
<em>Please make sure that this is an issue related to performance of TensorFlow.
As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:performance_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
When use the following function for FLOPS calculation on mac M1 
`def get_flops(model: Union[Model, Sequential], batch_size: Optional[int] = None) -> int:
    """
    Calculate FLOPS for tf.keras.Model or tf.keras.Sequential .
    Ignore operations used in only training mode such as Initialization.
    Use tf.profiler of tensorflow v1 api.
    """
    if not isinstance(model, (Sequential, Model)):
        raise KeyError(
            "model arguments must be tf.keras.Model or tf.keras.Sequential instanse"
        )

    if batch_size is None:
        batch_size = 1
        
    inputs = [tf.TensorSpec([batch_size] + inp.shape[1:], inp.dtype) for inp in model.inputs]
    
    # print("@@@@@@@@@@@@@",inputs)
    real_model = tf.function(model).get_concrete_function(inputs)
    # frozen_func, _ = convert_variables_to_constants_v2_as_graph(real_model)

    # Calculate FLOPS with tf.profiler
    run_meta = tf.compat.v1.RunMetadata()
    opts = tf.compat.v1.profiler.ProfileOptionBuilder.float_operation()
    flops = tf.compat.v1.profiler.profile(
        graph=real_model.graph, run_meta=run_meta, cmd="scope", options=opts
    )
    
    return flops.total_float_ops`

give wrong FLOPS as output: **2632160**

**Describe the expected behavior**

While the same code gives the correct calculation when tried on NVIDIA enabled machine. **VGGNET-16 FLOPS: 4587547104**
Expecting the same result to be calculated when done on Mac M1 machine.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by mohantym** [comment]
Hi @shujaat81 ! Could you please attach the above code as a Colab gist? Thank you!

**Comment by shujaat81** [comment]
Hi @mohantym Please find the link for colab
[https://colab.research.google.com/drive/1Hw3HuAa2vOEM6VxwnorV5pdToqx3mlJB?usp=sharing](url)

**Comment by mohantym** [comment]
@shujaat81 ! Could you please confirm your Tensorflow version too?

**Comment by shujaat81** [comment]
@mohantym Iam using 2.4.0-rc0 version

**Comment by mohantym** [comment]
Hi @Saduf2019 ! Could you please look at this issue? Attaching [gist](https://colab.sandbox.google.com/gist/mohantym/0c0f600648c497a7dec8ea499aa48fd9/untitled0.ipynb#scrollTo=XzJQ-fothZQk) for reference.
